streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -646,18 +648,39 @@ public Subscription subscription(final Set<String> topics) {
                     for (final TaskId id : state.activeTasks()) {
                         topicPartitions.addAll(partitionsForTask.get(id));
                     }
-
                     partitionsByHostState.put(hostInfo, topicPartitions);
                 }
             }
+        } else if (minReceivedMetadataVersion >= 4) {
+            for (final Map.Entry<UUID, ClientMetadata> entry : clientsMetadata.entrySet()) {
+                final HostInfo hostInfo = entry.getValue().hostInfo;
+
+                if (hostInfo != null) {
```

Comment:
could we also check for entry.getValue().state is not null

Reply:
This is not required as entry can't be null and still be in assign, entry.getValue().state is already being used in Step 2.